tests/requirements.in

```diff
@@ -1,5 +1,22 @@
-# This requirements file is used for stable ansible docs builds
+# This requirements file is used for doc builds
 # It depends on specific, tested antsibull-docs and sphinx versions
 
 -c constraints.in # <-- contains known limitations
--r requirements-relaxed.in # <-- contains base set of dependencies
+
+# Base set of dependencies.
+
+sphinx
+sphinx-intl # translation utility used by docs/docsite/Makefile
+sphinx-notfound-page
+sphinx-ansible-theme
+antsibull-docs
+sphinx-copybutton
+
+# Known limitations for indirect/transitive dependencies.
+
+jinja2 >= 3.0.0 # used by hacking/build_library/build_ansible/command_plugins/generate_man.py and dump_keywords.py
+pyyaml >= 5.1 # used by ansible-core
+resolvelib >= 0.5.3, < 1.1.0 # used by ansible-core
```

Comment:
I don't think jinja2, pyyaml, and resolvelib should be treated as indirect/transitive dependencies, since we (resp. ansible-core, which we copy in here) directly depend on them. IMO they are regular dependencies and should be in the "base set of dependencies" list.

Reply:
Thanks for catching this @felixfontein I'll update.

Reply:
@felixfontein Just pushed a commit to update this. Bear with me, but should we put the jinja2, pyyaml, and resolvelib pins in the constraints file too? I'm still not sure about this.

Reply:
So they should be in requirements, not in constraints.

Reply:
@felixfontein Thanks again for taking the time to explain. I still find it a little confusing as docs for constraints file seem to indicate that is the place to specify exact versions for packages. It's also the case that `sphinx` is a direct dependency for the build, so it still seems a little inconsistent to me. If it looks good to you though, I'm happy enough. I don't want to make this an ordeal. :smile:

Reply:
You could also split this up, and mention `jinja2`, `pyyaml`, and `resolvelib` in requirements, and add the version ranges in constraints.

Reply:
Ah, OK. Maybe I wasn't clear but that's what I was getting at. It seemed like the versions should be in the constraints file.

Reply:
@felixfontein Pushed a final commit with that change. Would you mind taking another look?